includes/tracks/events/class-wc-products-tracking.php

```diff
@@ -17,12 +17,38 @@ class WC_Products_Tracking {
 	 * Init tracking.
 	 */
 	public function init() {
+		add_action( 'load-edit.php', array( $this, 'track_products_view' ), 10 );
 		add_action( 'edit_post', array( $this, 'track_product_updated' ), 10, 2 );
 		add_action( 'transition_post_status', array( $this, 'track_product_published' ), 10, 3 );
 		add_action( 'created_product_cat', array( $this, 'track_product_category_created' ) );
 		add_action( 'add_meta_boxes_product', array( $this, 'track_product_updated_client_side' ), 10 );
 	}
 
+	/**
+	 * Send a Tracks event when the Products page is viewed.
+	 */
+	public function track_products_view() {
+		// We only record Tracks event when no `_wp_http_referer` query arg is set, since
+		// when searching, the request gets sent from the browser twice,
+		// once with the `_wp_http_referer` and once without it.
+		//
+		// Otherwise, we would double-record the view and search events.
+
+		// phpcs:disable WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.NonceVerification
+		if ( isset( $_GET['post_type'] )
+			&& 'product' === wp_unslash( $_GET['post_type'] )
+			&& ! isset( $_GET['_wp_http_referer'] ) ) {
+			// phpcs:enable
+
+			WC_Tracks::record_event( 'products_view' );
+
+			if ( isset( $_GET['s'] ) ) {
```

Comment:
Maybe check if `s` is empty instead, since it's easy to just hit the search button without any query.

Reply:
Done!